### PR TITLE
retdec 5.0 (new formula)

### PR DESCRIPTION
**Pull request by Ishaanahuja7**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/retdec.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      system "make", "install"
+    end
+  end
+
+  test do
+    assert_match "\#\#\#\#\# Decompiling",
+    shell_output("#{bin}/retdec-decompiler.py --no-memory-limit -o #{testpath}/a.c #{test_fixtures("mach/a.out")}")
+  end
+end
+--- a/deps/yara/patch.cmake
```
[the remark this answers is not in the document]
Comment (Ishaanahuja7) [reply Formula/retdec.rb:51]: RetDec builds its dependencies locally, don't see any cmake option for using an already installed dependency instead

File: Formula/retdec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  def install
+    inreplace "cmake/options.cmake", "set_if_at_least_one_set(RETDEC_ENABLE_OPENSLL
+		RETDEC_ENABLE_CRYPTO)", ""
+    inreplace "deps/CMakeLists.txt", "cond_add_subdirectory(openssl RETDEC_ENABLE_OPENSLL)", ""
+    inreplace "src/crypto/CMakeLists.txt", "ALIAS crypto)", "ALIAS crypto)
+
+    find_package(OpenSSL 1.1.1 REQUIRED)"
+    inreplace "src/crypto/CMakeLists.txt", "retdec::deps::openssl-crypto", "OpenSSL::Crypto"
+    inreplace "src/crypto/retdec-crypto-config.cmake", "openssl-crypto", ""
```
Comment (SMillerDev) [inline Formula/retdec.rb:33]: Why is this needed?

Comment (Ishaanahuja7) [reply Formula/retdec.rb:33]: Fix for finding OpenSSL from system so that it doesn't build it locally

Comment (Ishaanahuja7) [reply Formula/retdec.rb:33]: I found a fix for Mac M1: Update OpenSSL used in RetDec to 1.1.1i which supports Mac M1 (Using inreplace). RetDec 4.0 currently fetches 1.1.1c which doesn't support Mac M1. After this fix, RetDec will build OpenSSL locally and I can remove openssl dependency. Is that better than what I already have?

Comment (SMillerDev) [reply Formula/retdec.rb]: It won't be merged depending on a vendored version of openssl, see https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

If there is no acceptable release yet we'll have to wait for such a release.

Comment (Ishaanahuja7) [reply Formula/retdec.rb]: So my fix passed all checks. If vendored version is not acceptable then my previous fix for using openssl locally works then or no?

Comment (Ishaanahuja7) [reply Formula/retdec.rb]: I mean it finds the homebrew installed openssl 1.1.1 so its not vendored anymore right since its using latest homebrew openssl?

**Comment by Ishaanahuja7** [comment]
So which version should will get approved? The current one or 1 commit before one?

**Comment by Ishaanahuja7** [comment]
So which version will get approved? The current one or 1 commit before one?

**Comment by SMillerDev** [comment]
One where it builds against homebrew dependencies without patches.

File: Formula/retdec.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  homepage "https://github.com/avast/retdec"
+  license all_of: ["MIT", "Zlib"]
+  head "https://github.com/avast/retdec.git", branch: "master"
+
+  stable do
+    url "https://github.com/avast/retdec.git",
+      branch:   "master",
+      revision: "8503282c6d1295ba58ed5fed4ccb5605ed3febfd"
+    version "5.0"
+  end
```
Comment (SMillerDev) [inline Formula/retdec.rb:12]: It should be using tagged versions

File: Formula/retdec.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  depends_on "openssl@1.1"
+  depends_on "python@3.10"
+
+  on_macos do
+    depends_on xcode: :build
+    depends_on macos: :mojave
+  end
+
+  on_linux do
+    depends_on "gcc@7"
```
Comment (SMillerDev) [inline Formula/retdec.rb:28]: Why 7?

Comment (Ishaanahuja7) [reply Formula/retdec.rb:28]: For C++17 support

File: Formula/retdec.rb (hunk 3 above)
Comment (Ishaanahuja7) [reply Formula/retdec.rb:12]: They only have tagged versions for releases (Which would require patches for OpenSSL)

Comment (SMillerDev) [reply Formula/retdec.rb:12]: So we need to wait for a release that includes this. People have lived without retdec in Homebrew/core for a while. They can wait a little longer.

Comment (Ishaanahuja7) [reply Formula/retdec.rb:12]: Okay, let's keep this pull request open until new release then?

File: Formula/retdec.rb (hunk 4 above)
Comment (SMillerDev) [reply Formula/retdec.rb:28]: Let's just do `gcc` then.

Comment (Ishaanahuja7) [reply Formula/retdec.rb]: Okay done

File: Formula/retdec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  homepage "https://github.com/avast/retdec"
+  license all_of: ["MIT", "Zlib"]
+  head "https://github.com/avast/retdec.git", branch: "master"
+
+  stable do
+    url "https://github.com/avast/retdec.git",
+      branch:   "master",
+      revision: "0749a46b2490c8d499d64f08629271e16c311d82"
+    version "5.0"
+  end
```
Comment (SMillerDev) [inline Formula/retdec.rb:12]: This shouldn't be a separate block

File: Formula/retdec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+  fails_with gcc: "5"
+
+  def install
+    gcc = Formula["gcc"] if OS.linux?
+
+    cmake_args = std_cmake_args
+    cmake_args << "-DCMAKE_C_COMPILER=#{gcc.opt_bin}/gcc-#{gcc.version.major}" if OS.linux?
+    cmake_args << "-DCMAKE_CXX_COMPILER=#{gcc.opt_bin}/g++-#{gcc.version.major}" if OS.linux?
```
Comment (SMillerDev) [inline Formula/retdec.rb:38]: This shouldn't need to be specified

File: Formula/retdec.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+    mkdir "build" do
+      system "cmake", "..", *cmake_args
+      system "make", "install"
+    end
```
Comment (SMillerDev) [inline Formula/retdec.rb:43]:
```suggestion
    system "cmake", "-S", ".", "-B", "build", *args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```
